source/content/guides/composer-convert.md

```diff
@@ -1,156 +1,301 @@
 
-```{promptUser:user}
-composer require drupal/MODULE_NAME:^VERSION
-```
+  This command works on Drush 8. If you're using Drush 9, use `pm:list` or refer to [Drush Commands](https://drushcommands.com/drush-9x/pm/pm:projectinfo/):
 
-Where `MODULE_NAME` is the machine name of the module in question, and `VERSION` is the version of that module your site is currently using. Composer may pull in a newer version than what you specify, depending upon what versions are available. You can read more about the caret (`^`) [in the documentation for Composer](https://getcomposer.org/doc/articles/versions.md#caret-version-range-) itself.
+  ```bash{promptUser:user}
+  terminus drush $SITE.dev -- pm:projectinfo --fields=name,version --format=table
+  ```
 
-If you get the following error:
+  This will list each module followed by the version of that module that is installed.
 
-```none
-[InvalidArgumentException]
-Could not find a version of package drupal/MODULE_NAME matching your minimum-stability (stable). Require it with an explicit version constraint allowing its desired stability.
-```
+1. You can add these modules to your new codebase using Composer by running the following for each module in the `$SITE-composer` directory:
+
+  ```bash{promptUser:user}
+  composer require drupal/MODULE_NAME:^VERSION
+  ```
+
+  Where `MODULE_NAME` is the machine name of the module in question, and `VERSION` is the version of that module the site is currently using. Composer may pull in a newer version than what you specify, depending upon what versions are available. You can read more about the caret (`^`) in the [Composer documentation](https://getcomposer.org/doc/articles/versions.md#caret-version-range-).
+
+   - If you get the following error, the module listed in the error (or its dependencies) does not meet compatibility requirements:
+
+   ```none
+   [InvalidArgumentException]
+   Could not find a version of package drupal/MODULE_NAME matching your minimum-stability (stable). Require it with an explicit version constraint allowing its desired stability.
+   ```
+
+   If there is no stable version you can switch to, you may need to adjust the `minimum-stability` setting of `composer.json` to a more relaxed value, such as `beta`, `alpha`, or even `dev`. You can read more about `minimum-stability` in the [Composer documentation](https://getcomposer.org/doc/04-schema.md#minimum-stability).
+
+**Trust a robot?**
+
+One of Pantheon's engineers got tired of doing this process by hand, so he trained some robots to identify modules and put them into `composer.json`.
+
+Robots are cool, but they're not perfect, so you should understand the goal of this process as well as the limitations of automating the process.
+
+<Accordion title="A script that can help add modules to composer.json" id="modules-script" icon="cogs">
+
+First, disclaimers:
+
+Pantheon support is not available for this script or anything that results from trying to use it.
```

Comment:
I think we need slightly higher levels of support for scripts we publish in official documentation. This statement sounds a little too strong. Do we have precedence for this (support or lack of support for scripts in documentation)?

I think we should publish the docs without the scripts first & then have a discussion about support. e.g. with nominal buy-in from Devrel we could perhaps direct folks to #composer-workflow

Reply:
Regarding precedence, we have this text in Terminus plugins:

> This repository is provided without warranty or direct support. Issues and questions may be filed in GitHub but their resolution is not guaranteed.